src/main/java/org/openrewrite/staticanalysis/URLEqualsHashCode.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import org.openrewrite.java.template.RecipeDescriptor;
+
+import java.net.URI;
+import java.net.URL;
+
+@SuppressWarnings("UrlHashCode")
+@RecipeDescriptor(
+        name = "URL Equals and Hash Code",
+        description = "Uses of equals() and hashCode() cause java.net.URL to make blocking internet connections. " +
+                      "Instead, use java.net.URI.",
+        tags = {"errorprone"}
+)
```

Comment:
I think we should add some Markdown formatting here. Also, I don't think the `errorprone` tag is very useful. Does Errorprone have a check of this type?

Reply:
https://errorprone.info/bugpattern/URLEqualsHashCode

I'll add the markdown formatting. Good point. 😄